sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -219,29 +242,57 @@ def incr_mean_variance_axis0(X, last_mean, last_var, unsigned long last_n):
     `utils.extmath._batch_mean_variance_update`.
 
     """
-    cdef unsigned long n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
+    if X.dtype != np.float32:
+        X = X.astype(np.float64)
+    return _incr_mean_variance_axis0(X.data, X.shape, X.indices, X.indptr,
+                                     last_mean, last_var, last_n)
+
+
+def _incr_mean_variance_axis0(np.ndarray[floating, ndim=1] X_data,
```

Comment:
Ah sorry for the my carelessness, you are right! Thanks @ogrisel .
However, in my merged PR #6588 and #6539, I didn't use `cdef`, which seems to be a wrong move.

Should I fix that in this PR or create another PR?